### PR TITLE
Fix proto patching hack

### PR DESCRIPTION
**Pull request by ostromart**
This cleanup removes the troublesome go structs patch step and replaces it with synthetic types. 
This is required to correctly auto-generate the docs from the proto.

### CODE REVIEW

File: pkg/apis/istio/v1alpha2/istiocontrolplane_types.proto [hunk 1]
```diff
@@ -552,6 +556,15 @@ message ObjectMeta {
...
 }
 
+// GOTYPE: map[string]interface{}
+message TypeMapStringInterface {}
+
+// GOTYPE: interface{}
+message TypeInterface {}
+
+// GOTYPE: *IntOrStringForPB
+message IntOrStringForPB {}
```
Comment (ayj) [inline pkg/apis/istio/v1alpha2/istiocontrolplane_types.proto]: `TypeIntOrString` to match the other synthetic types?

Comment (ostromart) [reply pkg/apis/istio/v1alpha2/istiocontrolplane_types.proto]: Done.

File: pkg/apis/istio/v1alpha2/istiocontrolplane_types.proto [hunk 2]
```diff
@@ -544,6 +546,8 @@ message DeploymentStrategy {
 
 // Mirrors k8s.io.api.apps.v1.RollingUpdateDeployment for unmarshaling.
 message RollingUpdateDeployment {
+    TypeIntOrStringForPB maxUnavailable = 1;
```
Comment (ayj) [inline pkg/apis/istio/v1alpha2/istiocontrolplane_types.proto]: ni: extra space here

Comment (ostromart) [reply pkg/apis/istio/v1alpha2/istiocontrolplane_types.proto]: Done.

**Review by ayj — APPROVED** [review]

**Pull request merged**